src/_utils.ts

```diff
@@ -158,3 +158,19 @@ export function doesDependencyExist(
     return false;
   }
 }
+
+export function sanitizePackageManagerName(name: string): {
+  name: any;
+  warning: string | undefined;
+} {
+  const sanitized = /\w+$/.exec(name)?.[0];
+
+  if (sanitized && name !== sanitized) {
+    return {
+      name: sanitized,
+      warning: `Abnormal characters found in \`packageManager\` field, sanitizing from \`'${name}'\` to \`'${sanitized}'\``,
+    };
+  }
+
+  return { name, warning: undefined };
```

Comment:
Returning `name: sanitized` was fine. For consistency we could use `warnings` btw.